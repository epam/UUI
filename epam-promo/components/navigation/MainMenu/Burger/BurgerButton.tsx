import { withMods, IHasIcon, IDropdownToggler } from '@epam/uui-core';
import { Button, ButtonProps } from '@epam/uui-components';
import { ReactComponent as SvgTriangle } from '../../../../icons/triangle.svg';
import css from './BurgerButton.scss';

export interface BurgerButtonMods extends IHasIcon, IDropdownToggler {
    type?: 'primary' | 'secondary';
    indentLevel?: number;
}

export const BurgerButton = withMods<ButtonProps, BurgerButtonMods>(
    Button,
    props => [
        css.root,
        css['button-' + (props.type || 'primary')],
        css['indent-' + (props.indentLevel || 0)],
        props.isDropdown && css.dropdown,
        props.icon && css.hasIcon,
    ],
    () => ({ dropdownIcon: SvgTriangle, dropdownIconPosition: 'left', role: 'menuitem' })
);
