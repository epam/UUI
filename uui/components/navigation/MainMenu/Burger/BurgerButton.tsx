import { withMods, IHasIcon, IDropdownToggler } from '@epam/uui-core';
import { Button, ButtonProps } from '@epam/uui-components';
import { ReactComponent as SvgTriangle } from '@epam/assets/icons/navigation-arrow_down-outline.svg';
import css from './BurgerButton.module.scss';

interface BurgerButtonMods extends IHasIcon, IDropdownToggler {
    /*
    * Defines component type.
    */
    type?: 'primary' | 'secondary';
    /*
    * Defines components' indent.
    */
    indentLevel?: number;
}

export const BurgerButton = /* @__PURE__ */withMods<ButtonProps, BurgerButtonMods>(
    Button,
    (props) => [
        css.root,
        'uui-main_menu-burger-button',
        css['button-' + (props.type || 'primary')],
        css['indent-' + (props.indentLevel || 0)],
        props.isDropdown && css.dropdown,
        props.icon && css.hasIcon,
    ],
    () => ({ dropdownIcon: SvgTriangle, dropdownIconPosition: 'left', role: 'menuitem' }),
);
