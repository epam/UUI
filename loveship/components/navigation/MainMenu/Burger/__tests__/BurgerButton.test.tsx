import React from 'react';
import renderer from 'react-test-renderer';
import { BurgerButton } from '../BurgerButton';
import { ReactComponent as AcceptIcon } from '../../../../icons/accept-12.svg';

describe('BurgerButton', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<BurgerButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<BurgerButton caption="Test button" icon={AcceptIcon} href="#" target="_blank" type="secondary" isDropdown isOpen={false} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
