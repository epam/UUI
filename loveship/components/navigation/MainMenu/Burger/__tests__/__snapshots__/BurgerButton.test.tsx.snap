// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerButton should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled root button-primary indent-0"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={-1}
  type="button"
/>
`;

exports[`BurgerButton should be rendered correctly with props 1`] = `
<a
  className="container uui-button-box uui-enabled -clickable root button-secondary indent-0 dropdown hasIcon"
  href="#"
  onClick={[Function]}
  role="link"
  tabIndex={0}
  target="_blank"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <div
    className="uui-caption"
  >
    Test button
  </div>
</a>
`;
